FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -5,38 +5,80 @@
 import CoreData
 
 public final class CoreDataFeedStore: FeedStore {
-	private static let modelName = "FeedStore"
+	private static let modelName = "CoreDataFeedStore"
+
 	private static let model = NSManagedObjectModel(name: modelName, in: Bundle(for: CoreDataFeedStore.self))
-	
+
 	private let container: NSPersistentContainer
 	private let context: NSManagedObjectContext
-	
+
 	struct ModelNotFound: Error {
 		let modelName: String
 	}
-	
+
 	public init(storeURL: URL) throws {
 		guard let model = CoreDataFeedStore.model else {
 			throw ModelNotFound(modelName: CoreDataFeedStore.modelName)
 		}
-		
+
 		container = try NSPersistentContainer.load(
 			name: CoreDataFeedStore.modelName,
 			model: model,
 			url: storeURL
 		)
 		context = container.newBackgroundContext()
 	}
-	
+
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		context.perform {
+			do {
+				if
+					let cache = try CoreDataFeedCache.fetch(context: self.context) {
+					let feed = self.mapToLocalFeed(feedItems: cache.feedItems)
+					completion(.found(feed: feed, timestamp: cache.timestamp))
+				} else {
+					completion(.empty)
+				}
+			} catch {
+				completion(.failure(error))
+			}
+		}
 	}
-	
+
+	private func deleteCache() throws {
+		try context.execute(CoreDataFeedCache.deleteRequest())
+	}
```

Comment:
There is only a single cache model stored at a time. So using a batch delete is not needed. Batch delete is for deleting multiple elements at a time.

Here's how you can delete a single item:

```suggestion
	private func deleteCache() throws {
		if let cache = try CoreDataFeedCache.fetch(context: context) {
			context.delete(cache)
		}
	}
```